### DISCUSSION TITLE
No bar padding on monocle mode

### DISCUSSION THREAD

**Discussion by LorenzoMioso** [bakkeby/dwm-flexipatch#187] [Ideas]
Hi,
if BAR_PADDING patch is enable, would be cool to remove bar padding when switching to monocle layout and restore it when
switching back to tile layout.

**Comment by bakkeby** [comment]
For something like this I one could have two separate functions to enable or disable bar padding (and perhaps one to toggle if that is desirable).

That way users could control this via keyboard shortcuts, something along these lines:

```c
    { MODKEY,                       XK_t,          setlayout,              {.v = &layouts[0]} },
    { MODKEY,                       XK_t,          enablebarpadding,       {0} },
    { MODKEY,                       XK_f,          setlayout,              {.v = &layouts[1]} },
    { MODKEY,                       XK_f,          enablebarpadding,       {0} },
    { MODKEY,                       XK_m,          setlayout,              {.v = &layouts[2]} },
    { MODKEY,                       XK_m,          disablebarpadding,      {0} },
```

Perhaps cumbersome to work with if you have many layouts and it would not work out of the box if you use something like cyclelayouts to set the arrangement or right-clicking on the layout symbol on the bar.

The complication with this idea is that, although it seems obvious, how do you define what the "monocle" layout is? Sure you have the `monocle` layout function, but would you also consider a monocle arrangement through the flextile-deluxe layout in this scenario?

I guess you would end up with something in the `setlayout` function to do something like this. Another thing to consider is that if the user has intentionally disabled bar padding and they change to the monocle layout then you would not want the bar padding to be enabled again when you move back to tile layout.
